src/Form/matching-types.d.ts

```diff
@@ -104,24 +104,28 @@ type SupportedSubTypes =
     | 'creditCards.expirationYear'
     | 'creditCards.expiration'
 
+type SupportedSubTypesOrString =
```

Comment:
It would be nice if we don't need the string part here.

Reply:
Would using a more specific type as proposed above solve it?

Reply:
I think we should address this in a follow up, the return type of the getInputSubtype method is
```
@returns {SupportedSubTypes | string}
```
Due to it returning SupportedMainTypes or the right hand side of the SupportedSubTypes aka "username" etc.

Perhaps there's a rationale to this that @shakyShane had etc?

Reply:
We could also pass around both types until we need to split it which would clean up function signatures more I bet.

Reply:
Agreed we would move this to a follow up tiny PR to reduce risk of a reversion of this one.